### PR TITLE
releases, toc: add 3.1.0 GA release notes

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Add 3.1 GA release notes.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/2777
- Other reference link(s):

### CODE REVIEW

File: releases/3.1.0-ga.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+TiDB version: 3.1.0 GA
+
+TiDB Ansible version: 3.1.0 GA
+
+## Compatibility Changes
+
++ TiDB
+
+    - Support directly stopping the start of TiDB if the HTTP listening port is unavailable, when the `report-status` configuration item is enabled [#16291](https://github.com/pingcap/tidb/pull/16291)
```
Comment (TomShawn) [inline releases/3.1.0-ga.md:18]:
```suggestion
    - Support directly stopping starting TiDB if the HTTP listening port is unavailable when the `report-status` configuration item is enabled [#16291](https://github.com/pingcap/tidb/pull/16291)
```

File: TOC.md [hunk 2]
```diff
@@ -394,6 +394,7 @@
     - [4.0.0-beta.1](/releases/4.0.0-beta.1.md)
     - [4.0.0-beta](/releases/4.0.0-beta.md)
   + v3.1
+    - [3.1.0-GA](/releases/3.1.0-ga.md)
```
Comment (TomShawn) [inline TOC.md:397]:
```suggestion
    - [3.1.0 GA](/releases/3.1.0-ga.md)
```

**Comment by TomShawn** [comment]
@ran-huang Please update `/reference/releases/rn.md`.

**Comment by ran-huang** [comment]
> @ran-huang Please update `/reference/releases/rn.md`.

@TomShawn Updated. PTAL

**Comment by TomShawn** [comment]
LGTM

File: releases/3.1.0-ga.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
++ TiKV
+
+    - Fix the panic issue caused by replica read [#7418](https://github.com/tikv/tikv/pull/7418) [#7369](https://github.com/tikv/tikv/pull/7369)
+    - Fix the issue that the restoration process creates empty Regions [#7419](https://github.com/tikv/tikv/pull/7419)
+    - Fix the issue that repeated resolve lock requests might harm the atomicity of pessimistic transactions [#7389](https://github.com/tikv/tikv/pull/7389)
+
++ TiFlash
+
+    - Fix the potential issue of the `rename table` operation when replicating schema from TiDB
+    - Disable the coarse-grained index filter by default
```
Comment (JaySon-Huang) [inline releases/3.1.0-ga.md:79]:
```suggestion
    - Disable the rough set filter by default
```

File: releases/3.1.0-ga.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
++ Placement Driver (PD)
+
+    - Optimize the hot Region scheduling [#2342](https://github.com/pingcap/pd/pull/2342)
+
++ TiFlash
+
+    - Add the metrics report related to the read and write workloads of DeltaTree engine
+    - Support pushing down the `fromUnixTime` and `dateFormat` functions
+    - Disable the coarse-grained index filter by default
```
Comment (ran-huang) [inline releases/3.1.0-ga.md:41]:
```suggestion
    - Disable the rough set filter by default
```

**Comment by TomShawn** [comment]
@scsldb @JaySon-Huang PTAL again

**Comment by scsldb** [comment]
LGTM

File: releases/3.1.0-ga.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    - Fix the issue of data loss caused by the `rename table` operation under multiple data path configurations
+    - Fix the issue that TiFlash reports incorrect storage space in some scenarios
+    - Fix the potential issue caused by reading from TiFlash when Region Merge is enabled
+
++ Tools
+
+    - TiDB Binlog
+
+        * Fix the issue that TiFlash-related DDL jobs might interrupt the replication of Drainer [#948](https://github.com/pingcap/tidb-binlog/pull/948) [#942](https://github.com/pingcap/tidb-binlog/pull/942)
+    - BR
```
Comment (lilin90) [inline releases/3.1.0-ga.md:89]:
```suggestion

    - BR

```

**Pull request merged**